### PR TITLE
wasm: Remove `wavm`

### LINKED ISSUE #32772: Remove `wavm`
Our `wavm` build compiles its own llvm libs

This requires a lot of ci resources, and breaks updating llvm

We could either unbreak the llvm update or potentially just use prebuilt llvm libs, but in offline conversations it was suggested we might not need the wavm build at all as we have other wasm engines compiled with llvm

### PR DESCRIPTION
**Pull request by phlax**
Fix #32772 

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: .azure-pipelines/pipelines.yml [hunk 1]
```diff
@@ -65,6 +65,9 @@ stages:
 # Presubmit/default
 - ${{ if eq(variables.pipelineDefault, true) }}:
   - template: stages.yml
+    parameters:
+      checkStageDeps:
```
Comment (phlax) [inline .azure-pipelines/pipelines.yml:69]: remove

**Review by htuch — COMMENTED** [review]
LGTM directionally, let me know when ready for full review.

**Review by RyanTheOptimist — COMMENTED** [review]
test/per_file_coverage.sh LGTM

**Review by htuch — APPROVED** [review]

**Pull request merged**